lib/phoenix/verified_routes.ex

```diff
@@ -108,32 +110,77 @@ defmodule Phoenix.VerifiedRoutes do
-  on different `~p`'s to handle each type.
+  If you find yourself needing to generate dynamic URLs which are defined statically
+  in the router, that's a good indicator you should refactor it into one or more
+  function, such as `posts_path/1` and `images_path/1`.
 
   Like any other compilation warning, the Elixir compiler will warn any time the file
-  that a ~p resides in changes, or if the router is changed. To view previously issued
-  warnings for files that lack new changes, the `--all-warnings` flag may be passed to
-  the `mix compile` task. For the following will show all warnings the compiler
-  has previously encountered when compiling the current application code:
+  that a `~p` resides in changes, or if the router is changed.
+
+  ## Localized routes and path prefixes
+
+  Applications that need to support internationalization (i18n) and localization (l10n)
+  often do so at the URL level. In such cases, there are different approaches one can
+  choose.
+
+  One option is to perform i18n at the domain level. You can have `example.com` (in which
+  you would detect the locale based on the "Accept-Language" HTTP header), `en.example.com`,
+  `en-GB.example.com` and so forth. In this case, you would have a plug that looks at the
+  host and at HTTP headers and calls `Gettext.get_locale/1` accordingly. The biggest benefit
+  of this approach is that you don't have to change the routes in your application and
+  verified routes works as is.
+
+  Some applications, however, like to the locale as part of the URL prefix:
+
+      scope "/:locale" do
+        get "/posts"
+        get "/images"
+      end
 
-      $ mix compile --all-warnings
+  For such cases, VerifiedRoutes allow you to configure a `path_prefixes` option, which
+  is a list of segments to prepend to the URL. For example:
 
-  *Note: Elixir >= 1.14.0 is required for comprehensive warnings. Older versions
-  will compile properly, but no warnings will be issued.
+      use Phoenix.VerifiedRoutes,
+        router: AppWeb.Router,
+        endpoint: AppWeb.Endpoint,
+        path_prefixes: [{Gettext, :get_locale, []}]
+
+  The above will prepend `"/#{Gettext.get_locale()}"` to every path and url generated with
+  `~p`. If your website has a handful of URLs that do not require the locale prefix, then
+  we suggest defining them in a separate module, where you use `Phoenix.VerifiedRoutes`
+  without the prefix option:
+
+      defmodule UnlocalizedRoutes do
+        use Phoenix.VerifiedRoutes,
+          router: AppWeb.Router,
+          endpoint: AppWeb.Endpoint,
+
+        # Since :path_prefixes was not declared,
+        # the code below won't prepend the locale and still be verified
+        def root, do: ~p"/"
+      end
+
+  Finally, for even more complex use cases, where the whole URL needs to localized,
+  see projects such as [Routex](https://github.com/BartOtten/routex) and
+  [CLDR.Routes](https://github.com/elixir-cldr/cldr_routes).
```

Comment:
Looks good @josé.  I think `ex_cldr_routes` (as known on hex) would be more accurate in this case than `CLDR.Routes`.